src/geo/gmaps/gmaps-layer-view-factory.js

```diff
@@ -0,0 +1,47 @@
+var cdb = require('cdb');
+var log = require('cdb.log');
+var GMapsBaseLayerView = require('./gmaps-base-layer-view');
+var LeafletWMSLayerView = require('../leaflet/leaflet-wms-layer-view');
+
+var GMapsLayerViewFactory = function () {};
+
+GMapsLayerViewFactory.prototype.createLayerView = function (layerModel, mapModel) {
+  var layerType = layerModel.get('type').toLowerCase();
+  var LayerViewClass;
+
+  if (layerType === 'tiled') {
```

Comment:
I decided to go with the simplest possible solution to raise a [discussion about factories](https://github.com/CartoDB/cartodb.js/pull/1008#discussion_r50563814). We were/are following different patterns in CartoDB.js (factory with map of types/constructors vs factory that provides a way to register mapping).